packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -310,10 +310,8 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
         const isDynamicImport = dynamicIndex >= 0
 
         // static import or valid string in dynamic import
-        const isLiteralId = !!specifier
-
         // If resolvable, let's resolve it
-        if (isLiteralId) {
+        if (!!specifier) {
```

Comment:
Is this change related to this PR? Could you reverted if not?

Reply:
It can be reverted, because if you extract it as a variable declaration, it will lose TypeScript control flow analysis.